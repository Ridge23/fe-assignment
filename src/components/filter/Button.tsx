import React from 'react';

interface IButton {
    caption: string;
}

import './sass/button.scss';

export default function Button({ caption }: IButton): JSX.Element {
    return <button className="button button--margin-sides">{caption}</button>
}