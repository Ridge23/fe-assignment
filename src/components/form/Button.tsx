import React from 'react';
import classNames from 'classnames';

interface IButton {
    caption: string;
    wide?: boolean;
    blue?: boolean;
}

import './sass/button.scss';

export default function Button({ caption, wide = false, blue = false }: IButton): JSX.Element {
    const buttonClass = classNames(
        'button', 
        'button--margin-sides',
        {
            'button--wide': wide,
            'button--blue': blue
        }
    );
    return <button className={buttonClass}>{caption}</button>
}