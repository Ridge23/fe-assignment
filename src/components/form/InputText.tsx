import React from 'react';

import './sass/input-text.scss';

interface IInputText {
    value: string;
    onChange: (value: string) => void;
}

export default function InputText({ value, onChange }: IInputText): JSX.Element {
    return <input 
        type="text" 
        value={value} 
        className="input-text" 
        onChange={(event) => { onChange(event.target.value) }}
    />
}