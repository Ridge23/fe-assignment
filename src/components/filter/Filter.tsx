import React, { useState } from 'react';

import InputText from '../form/InputText';
import Button from '../form/Button';

import './sass/filter.scss';

interface IFilter {
    handleChange: (value: string) => void;
}

export default function Filter({ handleChange }: IFilter): JSX.Element {
    const [value, setValue] = useState('');

    const changeHandler = (value: string = ''): void => { 
        setValue(value); 
        handleChange(value); 
    }

    return <div className="filter">
        <div className="filter__title">
            Filter
        </div>
        <div className="filter__content">
            <InputText 
                value={value} 
                onChange={changeHandler} 
            />
            <Button caption="Reset" onClick={() => { changeHandler('') }} />
        </div>
    </div>
}