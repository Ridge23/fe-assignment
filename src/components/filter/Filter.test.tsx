import 'jsdom-global/register';
import React from 'react';
import { mount } from 'enzyme';

import Filter from './Filter';
import InputText from '../form/InputText'
import Button from '../form/Button';

describe('Filter', () => {
    it('should render correctly', () => {
        const component = mount(<Filter handleChange={()=>{}} />);
        
        expect(component.childAt(0).hasClass('filter')).toBe(true);
        expect(component.contains(<input type="text" />));
        expect(component.contains(<InputText value='' onChange={() => {}} />));
        expect(component.contains(<Button caption='Reset' onClick={() => {}} />));

        expect(component).toMatchSnapshot();
    });
});