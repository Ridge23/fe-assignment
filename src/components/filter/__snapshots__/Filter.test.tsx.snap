// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter should render correctly 1`] = `
<Filter
  handleChange={[Function]}
>
  <div
    className="filter"
  >
    <div
      className="filter__title"
    >
      Filter
    </div>
    <div
      className="filter__content"
    >
      <InputText
        onChange={[Function]}
        value=""
      >
        <input
          className="input-text"
          onChange={[Function]}
          type="text"
          value=""
        />
      </InputText>
      <Button
        caption="Reset"
        onClick={[Function]}
      >
        <button
          className="button button--margin-sides"
          onClick={[Function]}
        >
          Reset
        </button>
      </Button>
    </div>
  </div>
</Filter>
`;
